sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -460,12 +446,15 @@ void Slab::freeChunk(void *Ptr) {
 bool Slab::hasAvail() { return NumAllocated != getNumChunks(); }
 
 auto Bucket::getAvailFullSlab() -> decltype(AvailableSlabs.begin()) {
+  // Return a slab that will be used for a single allocation.
   if (AvailableSlabs.size() == 0) {
     auto It = AvailableSlabs.insert(AvailableSlabs.begin(),
                                     std::make_unique<Slab>(*this));
     (*It)->setIterator(It);
   } else {
-    settings::CurPoolSize -= getSize();
+    // If a slab was available in the pool then note that the current pooled
+    // size has reduced by the size of this slab.
+    settings::CurPoolSize -= Size;
```

Comment:
What guarantees that there were `Size` bytes available in the slab?

Reply:
This Size is the size of all allocations in that bucket.